plugin_tests/dbs_mongo_test.py

```diff
@@ -0,0 +1,317 @@
+        self.assertIn('kitchen', resp.json['data'][0][1])
+        params['filters'] = json.dumps([{
+            'field': 'comments', 'operator': 'not_regex', 'value': 'kitchen'}])
+        resp = self.request(path='/file/%s/database/select' % (
+            self.dbFileId, ), user=self.admin, params=params)
+        self.assertStatusOk(resp)
+        self.assertNotIn('kitchen', resp.json['data'][0][1])
+        params['filters'] = json.dumps([{
+            'field': 'comments', 'operator': 'search', 'value': 'kItChen'}])
+        resp = self.request(path='/file/%s/database/select' % (
+            self.dbFileId, ), user=self.admin, params=params)
+        self.assertStatusOk(resp)
+        self.assertIn('kitchen', resp.json['data'][0][1].lower())
+        params['filters'] = json.dumps([{
+            'field': 'comments', 'operator': 'not_search', 'value': 'family'}])
+        resp = self.request(path='/file/%s/database/select' % (
+            self.dbFileId, ), user=self.admin, params=params)
+        self.assertStatusOk(resp)
+        self.assertNotIn('family', resp.json['data'][0][1].lower())
+        params['filters'] = json.dumps([{
+            'field': 'comments',
+            'operator': 'in',
+            'value': ['Fire Alarm', 'new boiler']}])
+        resp = self.request(path='/file/%s/database/select' % (
+            self.dbFileId, ), user=self.admin, params=params)
+        self.assertStatusOk(resp)
+        self.assertEqual(resp.json['data'][0][1], 'Fire Alarm')
+
+        params['sort'] = json.dumps(['zip', 'occupancytype'])
+        params['filters'] = json.dumps([{
+            'field': 'occupancytype', 'operator': 'is', 'value': '1-7FAM'}])
+        resp = self.request(path='/file/%s/database/select' % (
+            self.dbFileId, ), user=self.admin, params=params)
+        self.assertStatusOk(resp)
+        self.assertEqual(resp.json['data'][0][2], '1-7FAM')
+        params['filters'] = json.dumps([{
+            'field': 'occupancytype', 'operator': 'is', 'value': None}])
+        resp = self.request(path='/file/%s/database/select' % (
+            self.dbFileId, ), user=self.admin, params=params)
+        self.assertStatusOk(resp)
+        self.assertEqual(resp.json['data'][0][2], None)
+        params['filters'] = json.dumps([{
+            'field': 'occupancytype', 'operator': 'is_not', 'value': '1-7FAM'}])
+        resp = self.request(path='/file/%s/database/select' % (
+            self.dbFileId, ), user=self.admin, params=params)
+        self.assertStatusOk(resp)
+        self.assertNotEqual(resp.json['data'][0][2], '1-7FAM')
+        params['filters'] = json.dumps([{
+            'field': 'occupancytype', 'operator': 'not_is', 'value': None}])
+        resp = self.request(path='/file/%s/database/select' % (
+            self.dbFileId, ), user=self.admin, params=params)
+        self.assertStatusOk(resp)
+        self.assertNotEqual(resp.json['data'][0][2], None)
+
+    def XtestMongoDatabaseSelectClient(self):
```

Comment:
More that it was never added...

We don't currently have support for the concept of a clients in the Mongo connector.  In Postgres, this can be used to cancel a query when it is replaced by a fresher query (for when the user no longer cares about the answer to the first query).